BackEnd/utils/moderation/image_moderation.py

```diff
@@ -0,0 +1,49 @@
+from django.utils.timezone import now
+from utils.completeness_counter import completeness_count
+from utils.moderation.moderation_actions import ModerationActions
+
+
+class ModerationManager:
+    def __init__(self, profile):
+        self.profile = profile
+        self.moderation_is_needed = False
+        self.banner_logo = {"banner": None, "logo": None}
+
+    def update_status(self):
+        self.profile.status = self.profile.PENDING
+        self.profile.status_updated_at = now()
+        self.profile.save()
+        self.moderation_is_needed = True
+
+    def needs_moderation(self, image):
+        return image and not image.is_approved and not image.is_deleted
+
+    def check_for_moderation(self):
+        if self.needs_moderation(self.profile.banner):
+            self.update_status()
+            self.banner_logo["banner"] = self.profile.banner
+        if self.needs_moderation(self.profile.logo):
+            self.update_status()
+            self.banner_logo["logo"] = self.profile.logo
+        return self.moderation_is_needed
+
+    def handle_moderation_action(self, action):
+        moderation_action = ModerationActions(action)
+        if moderation_action == ModerationActions.APPROVE:
+            self.approve_image()
+        elif moderation_action == ModerationActions.REJECT:
+            pass
```

Comment:
I strongly recommend to ensure that code fails explicitly in case if action for some reason was not recognized.

Reply:
...and in general — do not fear to fail. To fail with an exception is usually much better, than silently return `None` where we actually had to implement some logic, but it wasn't done.